apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -68,15 +74,28 @@ export default class LibraryPublisher extends React.Component {
     onShareTeacherLibrary: PropTypes.func
   };
 
-  state = {
-    publishState: PublishState.DEFAULT,
-    libraryName: libraryParser.suggestName(
-      this.props.libraryDetails.libraryName
-    ),
-    libraryDescription: this.props.libraryDetails.libraryDescription,
-    selectedFunctions: this.props.libraryDetails.selectedFunctions,
-    profaneWords: null
-  };
+  constructor(props) {
+    super(props);
+
+    const initialSelectedFunctions = props.libraryDetails.selectedFunctions;
```

Comment:
Nice fix here. Mind adding a comment for what's going on? Since the backend stuff happens in a different component, it's a bit difficult to understand why this is necessary as an outsider.